Index: conquestgl.c
===================================================================
RCS file: /home/jon/src/repository/conquest/conquestgl.c,v
retrieving revision 1.9
retrieving revision 1.11
diff -u -r1.9 -r1.11
--- conquestgl.c	14 Nov 2004 22:34:04 -0000	1.9
+++ conquestgl.c	27 Dec 2005 21:10:41 -0000	1.11
@@ -2,7 +2,7 @@
 
 /************************************************************************
  *
- * $Id$
+ * $Id$
  *
  * Copyright 1999-2004 Jon Trulson under the ARTISTIC LICENSE. (See LICENSE).
  ***********************************************************************/
@@ -53,7 +53,12 @@
 void printUsage()
 {
   printf("Usage: conquestgl [-m ][-s server[:port]] [-r recfile] [ -t ]\n");
-  printf("                  [ -M <metaserver> ] [ -u ]\n\n");
+  printf("                  [-f ][-g <geometry>] [ -u ]\n");
+  printf("                  [ -M <metaserver> ]\n\n");
+  printf("    -f               run in fullscreen mode\n");
+  printf("    -g <geometry>    specify intial window width/height.\n");
+  printf("                      Format is WxH (ex: 1024x768).\n");
+  printf("                      Default is 800x600.\n");
   printf("    -m               query the metaserver\n");
   printf("    -s server[:port] connect to <server> at <port>\n");
   printf("                      default: localhost:1701\n");
@@ -90,6 +95,44 @@
   return TRUE;
 }
 
+/* parse the geometry arg.  Ensure that dConf init w/h are only
+ *  set if the geom arg was reasonably valid. Format WxH (ex: 1024x768)
+ */
+static void parseGeometry(char *geom)
+{
+  char geomcpy[32];
+  char *ch;
+  int w, h;
+
+  if (!geom || !*geom)
+    return;
+
+  memset((void *)geomcpy, 0, 32);
+  strncpy(geomcpy, geom, 32 - 1);
+
+  if ((ch = strchr(geomcpy, 'x')) == NULL)
+    return;                     /* invalid */
+
+  *ch = 0;
+  ch++;
+
+  if (!*ch)
+    return;
+
+  w = abs(atoi(geomcpy));
+  h = abs(atoi(ch));
+
+  if (!w || !h)
+    return;
+
+  /* set it up */
+  dConf.initWidth = w;
+  dConf.initHeight = h;
+
+  return;
+}
+
+
 /*  conquest - main program */
 int main(int argc, char *argv[]) 
 {
@@ -121,10 +164,18 @@
 
   cInfo.remotehost = strdup("localhost"); /* default to your own server */
 
+  dspInitData();
+
   /* check options */
-  while ((i = getopt(argc, argv, "mM:s:r:tP:d:u")) != EOF)    /* get command args */
+  while ((i = getopt(argc, argv, "fmM:s:r:tP:d:ug:")) != EOF)    /* get command args */
     switch (i)
       {
+      case 'f':
+        dConf.fullScreen = TRUE;
+        break;
+      case 'g':                 /* to let '-geometry' slide by *HACK* */
+        parseGeometry(optarg);
+        break;
       case 'm':
         wantMetaList = TRUE;
         break;
Index: gldisplay.h
===================================================================
RCS file: /home/jon/src/repository/conquest/gldisplay.h,v
retrieving revision 1.7
retrieving revision 1.9
diff -u -r1.7 -r1.9
--- gldisplay.h	30 May 2005 08:22:47 -0000	1.7
+++ gldisplay.h	27 Dec 2005 21:10:42 -0000	1.9
@@ -1,6 +1,6 @@
 /************************************************************************
  *
- * $Id$
+ * $Id$
  *
  * Copyright 2003 Jon Trulson under the ARTISTIC LICENSE. (See LICENSE).
  ***********************************************************************/
@@ -37,6 +37,8 @@
 
   unsigned int flags; 
 
+  int fullScreen;
+  int initWidth, initHeight;    /* initial wxh geometry */
 } dspConfig_t;
 
 #ifdef NOEXTERN_DCONF
@@ -235,5 +237,6 @@
 void drawQuad(GLfloat x, GLfloat y, GLfloat w, GLfloat h, GLfloat z);
 void drawExplosion(GLfloat x, GLfloat y, int snum, int torpnum);
 
+void dspInitData(void);
 
 #endif /* _GLDISPLAY_H */
Index: GL.c
===================================================================
RCS file: /home/jon/src/repository/conquest/GL.c,v
retrieving revision 1.23
retrieving revision 1.25
diff -u -r1.23 -r1.25
--- GL.c	10 Aug 2005 15:51:37 -0000	1.23
+++ GL.c	27 Dec 2005 21:10:41 -0000	1.25
@@ -2,7 +2,7 @@
  *
  * Jon Trulson, 1/2003
  *
- * $Id$
+ * $Id$
  *
  * Copyright 2003 Jon Trulson under the ARTISTIC LICENSE. (See LICENSE).
  */
@@ -37,7 +37,6 @@
 #include "glmisc.h"
 #include "glfont.h"
 
-#define NOEXTERN
 #include "conquest.h"
 
 /* torp animation state */
@@ -918,13 +917,16 @@
 }
 
 
-static void dspInitData(void)
+void dspInitData(void)
 {
   memset((void *)&dConf, 0, sizeof(dspConfig_t));
 
+  dConf.inited = False;
+
+  dConf.fullScreen = FALSE;
+  dConf.initWidth = 800;
+  dConf.initHeight = 600;
   dConf.wX = dConf.wY = 0;
-  dConf.wW = 800;
-  dConf.wH = 600;
 
   memset((void *)&dData, 0, sizeof(dspData_t));
 
@@ -956,20 +958,19 @@
 #ifdef DEBUG_GL
   clog("uiGLInit: ENTER");
 #endif
-  memset(&ConqData, 0, sizeof(ConqData));
-
-  dConf.inited = False;
-
-  dspInitData();
 
   glutInit(argc, argv);
   glutInitDisplayMode(GLUT_DEPTH | GLUT_DOUBLE | GLUT_RGBA | GLUT_ALPHA);
 
   glutInitWindowPosition(0,0);
-  glutInitWindowSize(dConf.wW, dConf.wH);
+
+  glutInitWindowSize(dConf.initWidth, dConf.initHeight);
   
   dConf.mainw = glutCreateWindow(CONQUESTGL_NAME);
 
+  if(dConf.fullScreen)
+    glutFullScreen();
+
   glutKeyboardFunc       (charInput);
   glutSpecialFunc        (input);
   glutMouseFunc          (mouse);
Index: conquest.h
===================================================================
RCS file: /home/jon/src/repository/conquest/conquest.h,v
retrieving revision 1.2
retrieving revision 1.3
diff -u -r1.2 -r1.3
--- conquest.h	2 Nov 2003 20:53:00 -0000	1.2
+++ conquest.h	27 Dec 2005 21:10:41 -0000	1.3
@@ -1,7 +1,7 @@
 
 /************************************************************************
  *
- * $Id$
+ * $Id$
  *
  * Copyright 2003 Jon Trulson under the ARTISTIC LICENSE. (See LICENSE).
  ***********************************************************************/
@@ -9,49 +9,6 @@
 #ifndef CONQUEST_H
 #define CONQUEST_H
 
-#include <X11/X.h>
-#include <X11/Xlib.h>
-#include <X11/Intrinsic.h>
-
-#ifdef NOEXTERN
-#define C2EXTERN
-#else
-#define C2EXTERN extern
-#endif
-
 #define CONQUESTGL_NAME "ConquestGL"
 
-typedef struct _ConquestDataRec *ConquestDataPtr;
-
-typedef struct _ConquestDataRec {
-  Display *dpy;
-  XtAppContext app;
-  XtIntervalId timerId;
-  XtWorkProcId workId;
-
-  Bool rendering;		/* are we rendering? */
-
-  Widget toplevel;
-  
-  /* some colors */
-  Pixel whitePixel;
-  Pixel blackPixel;
-  Pixel redPixel;
-  Pixel yellowPixel;
-  Pixel greenPixel;
-  Pixel bluePixel;
-  Pixel cyanPixel;
-  Pixel magentaPixel;
-
-  Pixel defaultPixel;
-  Pixel infoPixel;
-  Pixel specialPixel;
-  Pixel statsepPixel;
-  
-} ConquestDataRec;
-
-
-C2EXTERN ConquestDataRec ConqData;
-
-#undef C2EXTERN
 #endif
Index: HISTORY
===================================================================
RCS file: /home/jon/src/repository/conquest/HISTORY,v
retrieving revision 1.41
retrieving revision 1.42
diff -u -r1.41 -r1.42
--- HISTORY	10 Aug 2005 15:51:38 -0000	1.41
+++ HISTORY	25 Dec 2005 20:23:59 -0000	1.42
@@ -9,6 +9,17 @@
 
                               Conquest HISTORY
 
+???
+
+        - add patch by Bill Patterson adding a '-f' fullscreen option
+          to conquestgl.
+
+        - fixed localtime() libc futex wait hang in conquestd by
+          moving drcheck() out of the signal handler.  Bad Juju.
+          Thanks to Bill Patterson for letting me have access to his
+          server to locate the problem and test a fix.
+
+
 Version 8.1.1 (stable)
 
         - during playback of a recording, honor the user config
