--- conquest-7.1.orig/conqutil.c
+++ conquest-7.1/conqutil.c
@@ -472,7 +472,7 @@
   if (errfd == NULL)
     {
       umask(007);
-      sprintf(errfile, "%s/%s", CONQHOME, C_CONQ_ERRLOG);
+      sprintf(errfile, "%s/%s", CONQSTATE, C_CONQ_ERRLOG);
       if (ConquestGID == ERR)
 	{
 	  fprintf(stderr, "conqutil: clog():  ConquestUID == ERR!\n");
--- conquest-7.1.orig/conf.c
+++ conquest-7.1/conf.c
@@ -58,7 +58,7 @@
   sysconf_AllowRefits = TRUE;
 
 				/* start building the filename */
-  sprintf(conf_name, "%s/%s", CONQHOME, SYSCONFIG_FILE);
+  sprintf(conf_name, "%s/%s", CONQETC, SYSCONFIG_FILE);
 
   if ((conf_fd = fopen(conf_name, "r")) == NULL)
     {
@@ -737,7 +737,7 @@
   char conf_name[BUFFER_SIZE];
   int i, j, n;
 
-  sprintf(conf_name, "%s/%s", CONQHOME, SYSCONFIG_FILE);
+  sprintf(conf_name, "%s/%s", CONQETC, SYSCONFIG_FILE);
   umask(002);
   unlink(conf_name);
 
--- conquest-7.1.orig/conf.h
+++ conquest-7.1/conf.h
@@ -24,7 +24,7 @@
 # define CEXTERN extern
 #endif
 
-#define SYSCONFIG_FILE "etc/conquestrc"	/* relative to CONQHOME */
+#define SYSCONFIG_FILE "conquestrc"	/* relative to CONQETC */
 #define CONFIG_FILE ".conquestrc"
 
 #define CONF_MAXCOMMENTS 50
--- conquest-7.1.orig/defs.h
+++ conquest-7.1/defs.h
@@ -42,19 +42,15 @@
 # define CONQUEST_GROUP "conquest"
 #endif
 
-#ifndef CONQHOME
-# define CONQHOME "/opt/conquest"
-#endif
-
 /* Some variables that used to be in conqfig.c - relative to conquest
- *  installation directory (CONQHOME)
+ *  installation directories
  */
 
-#define C_CONQ_CONQDRIV  "bin/conqdriv"
-#define C_CONQ_COMMONBLK "etc/conquest_common.img"
-#define C_CONQ_ERRLOG    "etc/conquest.log"
-#define C_CONQ_NEWSFILE  "etc/conqnews.doc"
-#define C_CONQ_HELPFILE  "etc/conquest.doc"
+#define C_CONQ_CONQDRIV  "conqdriv"
+#define C_CONQ_COMMONBLK "conquest_common.img"
+#define C_CONQ_ERRLOG    "conquest.log"
+#define C_CONQ_NEWSFILE  "conqnews.doc"
+#define C_CONQ_HELPFILE  "conquest.doc"
 
 
 /* HAS_SETITIMER - Define if your system supports setitimer/getitimer().
--- conquest-7.1.orig/conqcm.c
+++ conquest-7.1/conqcm.c
@@ -293,7 +293,7 @@
 
   coff = 0;
   
-  sprintf(cmnfile, "%s/%s", CONQHOME, C_CONQ_COMMONBLK);
+  sprintf(cmnfile, "%s/%s", CONQSTATE, C_CONQ_COMMONBLK);
 
 				/* verify it's validity */
   if (check_cblock(cmnfile, CMN_MODE, SIZEOF_COMMONBLOCK) == FALSE)
--- conquest-7.1.orig/Makefile.in
+++ conquest-7.1/Makefile.in
@@ -7,10 +7,20 @@
 #
 ###########################################################################
 
+# fun with autoconf
+prefix = @prefix@
+exec_prefix = @exec_prefix@
+bindir = @bindir@
+datadir = @datadir@
+libdir = @libdir@
+libexecdir = @libexecdir@
+localstatedir = @localstatedir@
+sysconfdir = @sysconfdir@
+DESTDIR =
+
+
 # User defineable options
 CONQGROUP = @CONQGROUP@
-CONQBIN = @prefix@/bin
-CONQHOME = @prefix@/conquest
 MAKE = make
 
 # Unixware: try -Ki486 or -Kpentium -
@@ -19,10 +29,9 @@
 
 LCFLAGS = 
 
-CFLAGS = @CFLAGS@ @DEFS@ $(LCFLAGS) -DCONQHOME=\"$(CONQHOME)\" -DCONQUEST_GROUP=\"$(CONQGROUP)\"
+CFLAGS = @CFLAGS@ @DEFS@ $(LCFLAGS) -DCONQUEST_GROUP=\"$(CONQGROUP)\" -DCONQSTATE=\"$(localstatedir)\" -DCONQLIBEXEC=\"$(libexecdir)\" -DCONQSHARE=\"$(datadir)\" -DCONQETC=\"$(sysconfdir)\"
 CONQUEST = conquest
 CONQAI = conqai
-CONQOPER = conqoper
 CONQDRIV = conqdriv
 CONQSTRAT = conqstrat
 
@@ -42,15 +51,6 @@
 	userauth.o options.o
 
 # For installation
-BCONQUEST = $(CONQHOME)/bin/conquest
-BCONQOPER = $(CONQHOME)/bin/conqoper
-BCONQDRIV = $(CONQHOME)/bin/conqdriv
-BCONQAI = $(CONQHOME)/bin/conqai
-BCONQSTRAT = $(CONQHOME)/bin/conqstrat
-HCONQHELP = $(CONQHOME)/etc/$(CONQHELP)
-HCONQNEWS = $(CONQHOME)/etc/$(CONQNEWS)
-HCONQREADME = $(CONQHOME)/etc/README
-
 CONQLB = @CONQLB@
 
 EXTLBS = @LIBS@ @LEXLIB@ @YACCLIB@
@@ -73,16 +73,14 @@
 	@echo "             Use 'gmake clean' before building if you are"
 	@echo "             lacking a 'makedepend' command."
 
-all: $(CONQLB) $(CONQUEST) $(CONQOPER) $(CONQAI) $(CONQDRIV) $(CONQSTRAT)
+all: $(CONQLB) $(CONQUEST) conqoper $(CONQAI) $(CONQDRIV) $(CONQSTRAT) 
 
 # program building  
 ##################
 
 conquest: conquest.o 
 	@-rm -f conquest
-	$(CC) -o conquest $(CFLAGS) conquest.o $(CURDIR)/$(CONQLB) $(EXTLBS)
-	@chgrp $(CONQGROUP) conquest
-	@chmod 2771 conquest
+	$(CC) -o conquest $(CFLAGS) conquest.o -lconquest -L. $(EXTLBS)
 
 conquest.o: conquest.c c_defs.h defs.h conqdef.h conqcom.h conqcom2.h
 	$(CC) $(CFLAGS) -c conquest.c
@@ -92,15 +90,11 @@
 
 conqdriv: conqdriv.o 
 	@-rm -f conqdriv
-	$(CC) -o conqdriv $(CFLAGS) conqdriv.o $(CURDIR)/$(CONQLB) $(EXTLBS)
-	@chgrp $(CONQGROUP) conqdriv
-	@chmod 2771 conqdriv
+	$(CC) -o conqdriv $(CFLAGS) conqdriv.o -lconquest -L. $(EXTLBS)
 
 conqai: conqai.o 
 	@-rm -f conqai
-	$(CC) -o conqai $(CFLAGS) conqai.o $(CURDIR)/$(CONQLB) $(EXTLBS)
-	@chgrp $(CONQGROUP) conqai
-	@chmod 2771 conqai
+	$(CC) -o conqai $(CFLAGS) conqai.o -lconquest -L. $(EXTLBS)
 
 conqai.o: conqai.c c_defs.h defs.h conqdef.h conqcom.h conqcom2.h conqdata.h
 	$(CC) $(CFLAGS) -c -DCONQAIMAIN conqai.c 
@@ -111,16 +105,14 @@
 
 conqoper: conqoper.o 
 	@-rm -f conqoper
-	$(CC) -o conqoper $(CFLAGS) conqoper.o $(CURDIR)/$(CONQLB) $(EXTLBS)
-	@chgrp $(CONQGROUP) conqoper
-	@chmod 2771 conqoper
+	$(CC) -o conqoper $(CFLAGS) conqoper.o -lconquest -L. $(EXTLBS)
 
 conqoper.o: conqoper.c c_defs.h defs.h conqdef.h conqcom.h conqcom2.h
 	$(CC) $(CFLAGS) -c conqoper.c 
 
 # we won't die if the following fails, since a default conqdata.h is supplied
 conqstrat:  y.tab.o lex.yy.o
-	-$(CC) -o conqstrat $(CFLAGS) lex.yy.o y.tab.o $(LIBS) $(CURDIR)/$(CONQLB) $(EXTLBS)
+	-$(CC) -o conqstrat $(CFLAGS) lex.yy.o y.tab.o $(LIBS) -lconquest -L. $(EXTLBS)
 
 y.tab.c: conqstrat.c.y
 	-$(YACC) -d conqstrat.c.y
@@ -144,135 +136,22 @@
 # Install it!
 #############################
 
-install: all $(CONQHOME) $(CONQHOME)/etc $(CONQHOME)/bin $(CONQHOME)/lib \
-	$(CONQHOME)/lib/$(CONQLB) $(BCONQUEST) $(BCONQOPER) $(BCONQDRIV)  \
-	$(HCONQHELP) $(HCONQNEWS) $(CONQHOME)/lib/$(CONQLB) @prefix@ \
-	$(CONQBIN) $(CONQHOME)/etc/conqrule $(BCONQSTRAT) $(BCONQAI) \
-	$(CONQHOME)/etc/conquestrc $(CONQBIN)/conquest $(CONQBIN)/conqoper \
-	$(CONQBIN)/conqstrat $(CONQBIN)/conqdriv $(CONQBIN)/conqai \
-	$(HCONQREADME)
-
-@prefix@:
-	mkdir -p @prefix@
-	chmod 775 @prefix@
-	chgrp $(CONQGROUP) @prefix@
-
-$(CONQBIN):
-	mkdir -p $(CONQBIN)
-	chmod 775 $(CONQBIN)
-
-$(CONQBIN)/conquest: $(BCONQUEST)
-	@-rm -f $(CONQBIN)/conquest
-	@-ln -s $(BCONQUEST) $(CONQBIN)/conquest
-
-$(CONQBIN)/conqoper: $(BCONQOPER)
-	@-rm -f $(CONQBIN)/conqoper
-	@-ln -s $(BCONQOPER) $(CONQBIN)/conqoper
-
-$(CONQBIN)/conqdriv: $(BCONQDRIV)
-	@-rm -f $(CONQBIN)/conqdriv
-	@-ln -s $(BCONQDRIV) $(CONQBIN)/conqdriv
-
-$(CONQBIN)/conqstrat: $(BCONQSTRAT)
-	@-rm -f $(CONQBIN)/conqstrat
-	@-ln -s $(BCONQSTRAT) $(CONQBIN)/conqstrat
-
-$(CONQBIN)/conqai: $(BCONQAI)
-	@-rm -f $(CONQBIN)/conqai
-	@-ln -s $(BCONQAI) $(CONQBIN)/conqai
-
-$(CONQHOME):
-	mkdir -p $(CONQHOME)
-	chmod 775 $(CONQHOME)
-	chgrp $(CONQGROUP) $(CONQHOME)
-
-$(CONQHOME)/lib/$(CONQLB): $(CONQLB)
-	cp $(CONQLB) $(CONQHOME)/lib/$(CONQLB)
-	chgrp $(CONQGROUP) $(CONQHOME)/lib/$(CONQLB)
-	chmod 644 $(CONQHOME)/lib/$(CONQLB)
-
-$(CONQHOME)/etc:
-	mkdir -p $(CONQHOME)/etc
-	chmod 775 $(CONQHOME)/etc
-	chgrp $(CONQGROUP) $(CONQHOME)/etc
-
-$(CONQHOME)/bin:
-	mkdir -p $(CONQHOME)/bin
-	chmod 775 $(CONQHOME)/bin
-	chgrp $(CONQGROUP) $(CONQHOME)/bin
-
-$(CONQHOME)/lib:
-	mkdir -p $(CONQHOME)/lib
-	chmod 775 $(CONQHOME)/lib
-	chgrp $(CONQGROUP) $(CONQHOME)/lib
-
-$(CONQHOME)/etc/conquestrc: $(BCONQOPER)
-	$(BCONQOPER) -C
-	chmod 664 $(CONQHOME)/etc/conquestrc
-	chgrp $(CONQGROUP) $(CONQHOME)/etc/conquestrc
-
-$(BCONQUEST): conquest
-	@-rm -f $(BCONQUEST)
-	$(CC) -o conquest $(CFLAGS) conquest.o $(CONQHOME)/lib/$(CONQLB) $(EXTLBS)
-	cp conquest $(BCONQUEST)
-	chgrp $(CONQGROUP) $(BCONQUEST)
-	chmod 2771 $(BCONQUEST)
-	@FILEPRIV@ -f plock,tshar,core $(BCONQUEST)
-
-$(BCONQOPER): conqoper
-	@-rm -f $(BCONQOPER)
-	$(CC) -o conqoper $(CFLAGS) conqoper.o $(CONQHOME)/lib/$(CONQLB) $(EXTLBS)
-	cp conqoper $(BCONQOPER)
-	chgrp $(CONQGROUP) $(BCONQOPER)
-	chmod 2771 $(BCONQOPER)
-	@FILEPRIV@ -f plock,tshar,core $(BCONQOPER)
-
-$(BCONQDRIV): conqdriv
-	@-rm -f $(BCONQDRIV)
-	$(CC) -o conqdriv $(CFLAGS) conqdriv.o $(CONQHOME)/lib/$(CONQLB) $(EXTLBS)
-	cp conqdriv $(BCONQDRIV)
-	chgrp $(CONQGROUP) $(BCONQDRIV)
-	chmod 2771 $(BCONQDRIV)
-	@FILEPRIV@ -f plock,tshar,core $(BCONQDRIV)
-
-$(BCONQAI): conqai
-	@-rm -f $(BCONQAI)
-	$(CC) -o conqai $(CFLAGS) conqai.o $(CONQHOME)/lib/$(CONQLB) $(EXTLBS)
-	cp conqai $(BCONQAI)
-	chgrp $(CONQGROUP) $(BCONQAI)
-	chmod 2770 $(BCONQAI)
-	@FILEPRIV@ -f plock,tshar,core $(BCONQAI)	
-
-$(BCONQSTRAT): conqstrat
-	@-rm -f $(BCONQSTRAT)
-	-$(CC) -o conqstrat $(CFLAGS) lex.yy.o y.tab.o $(LIBS) $(CONQHOME)/lib/$(CONQLB) $(EXTLBS)
-	-cp conqstrat $(BCONQSTRAT)
-	-chgrp $(CONQGROUP) $(BCONQSTRAT)
-	-chmod 751 $(BCONQSTRAT)
-
-$(CONQHOME)/etc/conqrule: conqrule
-	cp conqrule $(CONQHOME)/etc/conqrule
-	chgrp $(CONQGROUP) $(CONQHOME)/etc/conqrule
-	chmod 644 $(CONQHOME)/etc/conqrule
-
-$(HCONQREADME): README
-	cp README $(HCONQREADME)
-	chgrp $(CONQGROUP) $(HCONQREADME)
-	chmod 644 $(HCONQREADME)
-
-$(HCONQHELP): $(CONQHELP)
-	cp $(CONQHELP) $(HCONQHELP)
-	chgrp $(CONQGROUP) $(HCONQHELP)
-	chmod 644 $(HCONQHELP)
-
-$(HCONQNEWS): $(CONQNEWS)
-	cp $(CONQNEWS) $(HCONQNEWS)
-	chgrp $(CONQGROUP) $(HCONQNEWS)
-	chmod 644 $(HCONQNEWS)
+install: all 
+	mkdir -p $(DESTDIR)/$(bindir) $(DESTDIR)/$(libdir) $(DESTDIR)/$(datadir) $(DESTDIR)/$(libexecdir)
+	cp conquest conqoper conqai conqstrat $(DESTDIR)/$(bindir)
+	cp conqdriv $(DESTDIR)/$(libexecdir)
+	cp $(CONQLB) $(DESTDIR)/$(libdir)
+	cp conqrule conquest.doc conqnews.doc README $(DESTDIR)/$(datadir)
+	chmod 2775 $(DESTDIR)/$(bindir)/conquest
+	chmod 2775 $(DESTDIR)/$(bindir)/conqoper
+	chmod 2775 $(DESTDIR)/$(libexecdir)/conqdriv
+	chmod 2774 $(DESTDIR)/$(bindir)/conqai
+	chmod 755 $(DESTDIR)/$(bindir)/conqstrat
+	chmod 775 $(DESTDIR)/$(localstatedir)
 
 clean:
-	rm -f *.o $(CONQUEST) $(CONQOPER) $(CONQAI) $(CONQDRIV) \
-		$(CONQLB) $(CONQSTRAT) y.tab.[ch] lex.yy.[c]
+	rm -f *.o conquest conqoper conqai conqdriv conqstrat \
+		$(CONQLB) y.tab.[ch] lex.yy.[c]
 
 realclean:
 	$(MAKE) clean
--- conquest-7.1.orig/conqsvr42.c
+++ conquest-7.1/conqsvr42.c
@@ -523,7 +523,7 @@
   
   if (pid == 0)
     {				/* The child: aka "The Driver" */
-      sprintf(drivcmd, "%s/%s", CONQHOME, C_CONQ_CONQDRIV);
+      sprintf(drivcmd, "%s/%s", CONQLIBEXEC, C_CONQ_CONQDRIV);
       execl(drivcmd, drivcmd, NULL);
       clog("drcreate(): exec(): %s", sys_errlist[errno]);
       perror("exec");		/* shouldn't be reached */
@@ -632,7 +632,7 @@
   char buf[MSGMAXLINE];
   char helpfile[BUFFER_SIZE];
   
-  sprintf(helpfile, "%s/%s", CONQHOME, C_CONQ_HELPFILE);
+  sprintf(helpfile, "%s/%s", CONQSHARE, C_CONQ_HELPFILE);
   sprintf( buf, "%s: Can't open.", helpfile );
   pagefile( helpfile, buf);
   
@@ -733,7 +733,7 @@
   char newsfile[BUFFER_SIZE];
   extern char *c_conq_newsfile;
   
-  sprintf(newsfile, "%s/%s", CONQHOME, C_CONQ_NEWSFILE);
+  sprintf(newsfile, "%s/%s", CONQSHARE, C_CONQ_NEWSFILE);
   
   pagefile( newsfile, "No news is good news.");
   
